jmh-tests/src/main/java/org/eclipse/collections/impl/jmh/FunctionalInterfaceSetTest.java

```diff
@@ -323,9 +324,9 @@ public void test_parallel_lazy_jdk()
     @Benchmark
     public MutableSet<Integer> serial_eager_ec()
     {
-        FastList<Integer> select1 = this.integersEC.select(each -> each % 10_000 != 0);
-        FastList<String> collect1 = select1.collect(String::valueOf);
-        FastList<Integer> collect2 = collect1.collect(Integer::valueOf);
+        MutableList<Integer> select1 = this.integersEC.select(each -> each % 10_000 != 0);
+        MutableList<String> collect1 = select1.collect(String::valueOf);
+        MutableList<Integer> collect2 = collect1.collect(Integer::valueOf);
```

Comment:
When I pulled some methods up to interfaces, I deleted some overrides that returned concrete classes. This is debatably the correct thing to do, but also is a not-totally-necessary source-breaking change that I could revert.

Reply:
Seems to be a theme here: do we want to mentally reinforce that collections return their own types or not? I think previous to this, the answer has been "yes."

Reply:
Implementations almost always return the same time, except some adapters. Return types have been inconsistent in interfaces. Plus only some methods return the same type. Filtering methods (select, reject, partition, selectInstancesOf) do return the same time. Transforming methods (collect, flatCollect) only sometimes return the same type. Often this is obvious from the signature. MapIterable.collect(Function) returns RichIterable, for example. But when an interface method returns that same interface, I guess the assumption is it's also going to be the same concrete type.